packages/@sanity/types/src/validation/types.ts

```diff
@@ -1,9 +1,289 @@
 import {Path} from '../paths'
+import {SchemaType} from '../schema'
+import {SanityDocument} from '../documents'
+import {ValidationMarker} from '../markers'
 
-export interface ValidationError {
+export type RuleTypeConstraint = 'Array' | 'Boolean' | 'Date' | 'Number' | 'Object' | 'String'
+export type FieldRules = {[fieldKey: string]: (fieldRule: Rule) => Rule}
+
+// Note: `RuleClass` and `Rule` are split to fit the current `@sanity/types`
+// setup. Classes are a bit weird in the `@sanity/types` package because classes
+// create an actual javascript class while simultaneously creating a type
+// definition.
+//
+// This implicitly creates two types:
+// 1. the instance type — `Rule` and
+// 2. the static/class type - `RuleClass`
+//
+// The `RuleClass` type contains the static methods and the `Rule` instance
+// contains the instance methods. Downstream in the validation package, the Rule
+// implementation asserts the class declaration is of this type.
+export interface RuleClass {
+  FIELD_REF: symbol
+  array: (def?: SchemaType) => Rule
+  object: (def?: SchemaType) => Rule
+  string: (def?: SchemaType) => Rule
+  number: (def?: SchemaType) => Rule
+  boolean: (def?: SchemaType) => Rule
+  dateTime: (def?: SchemaType) => Rule
+
+  valueOfField: Rule['valueOfField']
+
+  new (typeDef?: SchemaType): Rule
+}
+
+export interface Rule {
```

Comment:
I think it's okay given it's visibility. if i were to guess what the import would be from `@sanity/types` while writing a schema in typescript, i would think it would just be `Rule`. happy to change this if you feel strongly about it though